### PR TITLE
Break out requests into own class, use requests.Session, UserAgent

### PR DESCRIPTION
**Pull request by christopherjwang**

### CODE REVIEW

File: tests/test_swiftype_enterprise_client.py [hunk 1]
```diff
@@ -249,14 +223,22 @@ def test_index_documents(self):
...
+            return stubbed_response
         # stub documents bulk_create endpoint
-        with patch('requests.post', return_value=bulk_create_docs_stubbed_response):
-            # stub document receipts bulk_show endpoint
-            with patch('requests.get', return_value=doc_receipts_stubbed_response):
-                self.assertEqual(
-                    self.client.index_documents(content_source_key, documents),
-                    doc_receipts_response
-                )
+        with patch('requests.Session.request', side_effect=endpoint_side_effect):
```
Comment (brianmcgue) [inline tests/test_swiftype_enterprise_client.py:236]: 👍  much cleaner

**Review by brianmcgue — APPROVED** [review]
LGTM

File: setup.py [hunk 2]
```diff
@@ -18,14 +18,19 @@
 finally:
     f.close()
 
+here = path.abspath(path.dirname(__file__))
+about = {}
+with open(path.join(here, 'swiftype_enterprise', '__version__.py'), 'r', 'utf-8') as f:
+    exec(f.read(), about)
```
Comment (brianmcgue) [inline setup.py:24]: I already approved this PR, but actually, will the version variables be assigned to the global or local scope?

```pycon
In [1]: v = '__testvar__ = 1'

In [2]: g, l = {}, {}

In [3]: exec(v, g, l)

In [4]: g['__testvar__']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-4-7cc76e349859> in <module>()
----> 1 g['__testvar__']

KeyError: '__testvar__'

In [5]: l['__testvar__']
Out[5]: 1
```

Comment (christopherjwang) [reply setup.py:24]: I copied that from [here](https://github.com/requests/requests/blob/master/setup.py#L53)

looks like [docs](https://docs.python.org/3/library/functions.html#exec) say: "If only globals is provided, it must be a dictionary, which will be used for both the global and the local variables." followed by "Remember that at module level, globals and locals are the same dictionary"

**Pull request merged**